File: 3392_count_subarrays_of_length_three_with_a_condition/Solution.cs
// LeetCode 3392 - Count Subarrays of Length Three With a Condition
// https://leetcode.com/problems/count-subarrays-of-length-three-with-a-condition/

public class Solution {
    public int CountSubarrays(int[] nums) {
        int ans = 0;
        for (int i = 0; i + 2 < nums.Length; i++) {
            if (nums[i] * 2 + nums[i + 2] * 2 == nums[i + 1]) ans++;
        }
        return ans;
    }
}

File: 3392_count_subarrays_of_length_three_with_a_condition/solution.py
# LeetCode 3392 - Count Subarrays of Length Three With a Condition
# https://leetcode.com/problems/count-subarrays-of-length-three-with-a-condition/

from typing import List


class Solution:
    def countSubarrays(self, nums: List[int]) -> int:
        ans = 0
        for i in range(len(nums) - 2):
            if nums[i] * 2 + nums[i + 2] * 2 == nums[i + 1]:
                ans += 1
        return ans
